mpu6050/mpu6050.go

```diff
@@ -1,95 +1,232 @@
-	data := make([]byte, 6)
-	d.bus.ReadRegister(uint8(d.Address), GYRO_XOUT_H, data)
-	// First the value is converted from a pair of bytes to a signed 16-bit
-	// value and then to a signed 32-bit value to avoid integer overflow.
-	// Then the value is scaled to µ°/s (micro-degrees per second).
-	// This is done in the following steps:
-	// 1. Multiply by 250 * 1000_000
-	// 2. Divide by 32768
-	// The following calculation (x * 15625 / 2048 * 1000) is essentially the
-	// same but avoids overflow. First both operations are divided by 16 leading
-	// to multiply by 15625000 and divide by 2048, and then part of the multiply
-	// is done after the divide instead of before.
-	x = int32(int16((uint16(data[0])<<8)|uint16(data[1]))) * 15625 / 2048 * 1000
-	y = int32(int16((uint16(data[2])<<8)|uint16(data[3]))) * 15625 / 2048 * 1000
-	z = int32(int16((uint16(data[4])<<8)|uint16(data[5]))) * 15625 / 2048 * 1000
-	return
+// you would get a value close to 6.3 radians (360 degrees).
+func (d *Device) AngularVelocity() (gx, gy, gz int32) {
+	const angvelOffset = 8
+	_ = d.RawData[angvelOffset+5] // This line fails to compile if RawData is too short.
+	gx = int32(convertWord(d.RawData[angvelOffset+0:])) * 4363 / 8192 * d.gRange
+	gy = int32(convertWord(d.RawData[angvelOffset+2:])) * 4363 / 8192 * d.gRange
+	gz = int32(convertWord(d.RawData[angvelOffset+4:])) * 4363 / 8192 * d.gRange
+	return gx, gy, gz
 }
 
-// SetClockSource allows the user to configure the clock source.
-func (d Device) SetClockSource(source uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), PWR_MGMT_1, []uint8{source})
+// Temperature returns the temperature of the device in milli-centigrade.
+func (d *Device) Temperature() (Celsius int32) {
+	const tempOffset = 6
+	return 1506*int32(convertWord(d.RawData[tempOffset:]))/512 + 37*1000
+}
+
+func convertWord(buf []byte) int16 {
+	return int16(binary.BigEndian.Uint16(buf))
+}
+
+// SetSampleRate sets the sample rate for the FIFO,
+// register ouput and DMP. The sample rate is determined
+// by:
+//
+//	SR = Gyroscope Output Rate / (1 + srDiv)
+//
+// The Gyroscope Output Rate is 8kHz when the DLPF is
+// disabled and 1kHz otherwise. The maximum sample rate
+// for the accelerometer is 1kHz, if a higher sample rate
+// is chosen, the same accelerometer sample will be output.
+func (p *Device) SetSampleRate(srDiv byte) (err error) {
+	// setSampleRate
+	var sr [1]byte
+	sr[0] = srDiv
+	if err = p.write8(_SMPRT_DIV, sr[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetClockSource configures the source of the clock
+// for the peripheral.
+func (p *Device) SetClockSource(clkSel byte) (err error) {
+	// setClockSource
+	var pwrMgt [1]byte
+
+	if err = p.read(_PWR_MGMT_1, pwrMgt[:]); err != nil {
+		return err
+	}
+	pwrMgt[0] = (pwrMgt[0] & (^_CLK_SEL_MASK)) | clkSel // Escribo solo el campo de clk_sel
+	if err = p.write8(_PWR_MGMT_1, pwrMgt[0]); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Is this actually useful? Can you give me an example where this would be used in practice?

If not, I suggest removing it to declutter the API.

(Also, it contains a Spanish comment?)